Remove requirement of JavaScript

dracos opened the pull request:
Fixes #8.
Only outstanding issue I can see is on `/datasets/` the full list of categories is shown but clicking them does not then only show the datasets in that category as it does with JS enabled. The 'easiest' way to fix in Jekyll would be to have a category collection, but would then involve listing all categories as files, whereas currently you only have to specify them in the datasets. Alternatively, a small plugin could generate them, but then it would have to be generated off-GHPages (Travis etc) before being uploaded. Alternatively alternatively, the category list could be hidden unless JS, being considered an 'enhancement', though ideally category pages would be a thing. It's still far better than it was.

dracos committed:
Use standard script/server location.

dracos committed:
Individual dataset pages viewable without JS.

The data was on the page, it was just hidden by default...

dracos committed:
Source pages now show data without JavaScript.

dracos committed:
Datasets page works without JavaScript.

And consequential amendments.

dracos committed:
Fix wrong if statement.

dracos committed:
Upgrade to GitHub Pages gem.

dracos committed:
Only show search filter if JS enabled.

dracos committed:
Icing on cake, no flash of content change in Categories sidebar.

dracos committed:
Icing on cake, no flash of content change in Categories sidebar.

dracos committed:
squash! Datasets page works without JavaScript.

Simpler query.

ajparsons commented:
@dracos : I think a categories collection is the way to go (adding categories should be a less frequent occurrence than adding or modifying data and it requires some manual work anyway - so easy enough to put a note in that file to remind people to run the 'convert to categories' plugin offline). 

I've got a version of that working that uses static pages for the categories - my current dilemma is if I should turn off the dynamic functionality of the datasets page to always send people to the static pages. I'm currently thinking that I should, but open to good reasons not to. 

Arguments for: 

Consistent URL - a link sent from a javascript to non-javascript user should work. 
Prevents javascript and non-javascript functionality getting out of step (if the categories page isn't created this should be noticeable if the updater has javascript on). 

Arguments against:

With improvements made to page loading, switching categories is now very fast. This would slow that down a little (but this is still quite fast and faster than we started with).
Divergence from JKAN core - changing the javascript will make future updates slightly trickier. But it's not a big change so I think this is manageable.

ajparsons committed:
Added script to create categories markdown files

From categories.yml + references in dataset files
markdown files are created in _categories.

This can then be treated as a collection to generate individual static files

ajparsons committed:
Commit dummy categories markdown files

ajparsons committed:
'\datasets' and category pages use same html to generate list of datasets

ajparsons committed:
Adjust link format to categories pages from index + datasets pages

ajparsons committed:
Show the current category as active in the category list

ajparsons committed:
Adjust JKAN javascript to not attempt to display content now rendered manually

ajparsons committed:
Add category collection into config, update readme

ajparsons committed:
Merge simpler query

dracos commented on _includes/datasets_display.html:
```diff
@@ -0,0 +1,72 @@
+
+{% comment %}Only display datasets in current category if there is a current category{% endcomment %}
+{% if current_category %}
+    {% assign current_slug = current_category | slugify %}
```
Instead of lines 4-12, why didn't you do `{% assign datasets = site.datasets | where:"category", current_category 
 %}`?

ajparsons replied:
It was a legacy from when I was passing that page the slug rather than the full name and so that didn't work, at some point I changed the process but didn't revisit that. But yep, that is much nicer, will make the change.

dracos commented on _includes/datasets_display.html:
```diff
@@ -0,0 +1,72 @@
+        {% endunless %}
+      {% endfor %}
+      <div class="js-hidden">
+      {% include filter-category.html categories=categories current=current_category  %}
```
This says `current` instead of `current_categoy`, but works because you don't need to mention variables if they're the same name, context is passed down (in fact, variables included here are found under `include.` so I don't think work as you think they do).

ajparsons replied:
I think I'm thinking of django templates. Will have a go through and either make explicit when I actually want to use includes, or take it out when the context can just be assumed.
